client/dashboard/me/notifications-extras/test/index.test.tsx

```diff
@@ -0,0 +1,370 @@
+	marketing: false,
+	research: false,
+	affiliates: false,
+	community: false,
+	promotion: false,
+	news: false,
+	digest: false,
+	reports: false,
+	news_developer: false,
+	wpcom_spain: false,
+	scheduled_updates: false,
+	learn: false,
+	a4a_agencies: false,
+	jetpack_agencies: false,
+	jetpack_manage_onboarding: false,
+	jetpack_marketing: false,
+	jetpack_research: false,
+	jetpack_promotion: false,
+	jetpack_news: false,
+	jetpack_reports: false,
+	akismet_marketing: false,
+	woopay_marketing: false,
+	gravatar_onboarding: false,
+};
+
+const defaultUserSettings: UserNotificationSettings = {
+	blogs: [],
+	other: {
+		timeline: { comment_like: false, comment_reply: false },
+		email: { comment_like: false, comment_reply: false },
+		devices: [],
+	},
+	wpcom: defaultWpcomSettings,
+};
+
+const mockGetNotificationSettingsApi = (
+	settings: UserNotificationSettings = defaultUserSettings
+) => {
+	return nock( 'https://public-api.wordpress.com:443' )
+		.get( '/rest/v1.1/me/notifications/settings' )
+		.reply( 200, settings );
+};
+
+const mockSaveNotificationSettingsApi = (
+	expectedSettings: Partial< WpcomNotificationSettings >
+) => {
+	return nock( 'https://public-api.wordpress.com:443' )
+		.post( '/rest/v1.1/me/notifications/settings', {
+			wpcom: expectedSettings,
+		} )
+		.reply( 200, { wpcom: expectedSettings } );
+};
+
+const notificationSnackBar = () => {
+	// Snackbar requires a custom matcher because its aria-live is not supported by the testing library
+	return document.getElementById( 'a11y-speak-polite' );
+};
+
+describe( 'NotificationsExtras', () => {
+	beforeEach( () => {
+		nock.disableNetConnect();
+		nock.cleanAll();
+		// Snackbar requires window.scrollTo to be defined
+		window.scrollTo = jest.fn();
+	} );
+
+	afterEach( () => {
+		nock.cleanAll();
+	} );
+
+	it( 'renders the page header and sections correctly', async () => {
+		mockGetNotificationSettingsApi();
+
+		dashboardRender(
+			<>
+				<Snackbars />
+				<NotificationsExtras />
+			</>
+		);
+
+		await waitFor( () => {
+			expect( screen.getByText( 'Extras' ) ).toBeVisible();
+		} );
+
+		expect(
+			screen.getByText( /Get curated extras like reports, digests, and community updates/ )
+		).toBeVisible();
+
+		expect( screen.getByText( 'Email from WordPress.com' ) ).toBeVisible();
+		expect( screen.getByText( 'Email from Jetpack' ) ).toBeVisible();
+
+		expect(
+			screen.getByText(
+				'Jetpack is a suite of tools connected to your WordPress site, like backups, security, and performance reports.'
+			)
+		).toBeVisible();
+	} );
+
+	it( 'displays all WordPress.com notification options', async () => {
+		mockGetNotificationSettingsApi();
+
+		dashboardRender(
+			<>
+				<Snackbars />
+				<NotificationsExtras />
+			</>
+		);
+
+		await waitFor( () => {
+			expect( screen.getAllByLabelText( 'Suggestions' )[ 0 ] ).toBeVisible();
+		} );
+
+		// WordPress.com options
+		expect( screen.getAllByLabelText( 'Suggestions' )[ 0 ] ).toBeVisible();
```

Comment:
You can use the `within` to scope your tests to a single component instance:

https://testing-library.com/docs/dom-testing-library/api-within/


```suggestion
         const wpCard = screen.getBy(.....card selector)
         
		expect( whithin(wpCard).getAllByLabelText( 'Suggestions' ) ).toBeVisible(); // Checks only inside a component.
```

Reply:
`within` makes this a bit more clean!